lbry/file_analysis.py

```diff
@@ -0,0 +1,338 @@
+import asyncio
+import json
+import logging
+import os
+import pathlib
+import re
+import shlex
+import shutil
+
+from lbry.conf import TranscodeConfig
+
+log = logging.getLogger(__name__)
+
+
+class VideoFileAnalyzer:
+    @staticmethod
+    def _matches(needles: list, haystack: list):
+        for needle in needles:
+            if needle in haystack:
+                return True
+        return False
```

Comment:
I don't think this method is necessary. You can then just do set operations:

instead of:
```
if not self._matches(container.split(","), ["webm", "mp4", "3gp", "ogg"])
```
do:
```
if not {"webm", "mp4", "3gp", "ogg"}.intersection(container.split(","))
```

Reply:
I like this intersection approach. Fixed and pushed.